src/main/java/seedu/address/storage/JsonAdaptedAppointment.java

```diff
@@ -0,0 +1,103 @@
+
+    /**
+     * Constructs a {@code JsonAdaptedPerson} with the given person details.
+     */
+    @JsonCreator
+    public JsonAdaptedAppointment(@JsonProperty("id") Integer id, @JsonProperty("doctor") String doctor,
+                                   @JsonProperty("patient") String patient, @JsonProperty("date") String date,
+                                   @JsonProperty("time") String time) {
+        this.id = id;
+        this.doctorName = doctor;
+        this.patientName = patient;
+        this.date = date;
+        this.time = time;
+    }
+
+    /**
+     * Converts a given {@code Appointment} into this class for Jackson use.
+     */
+    public JsonAdaptedAppointment(Appointment source) {
+        id = source.getId();
+        doctorName = source.getDoctorName();
+        patientName = source.getPatientName();
+        date = source.getDate().toString();
+        time = source.getTime().toString();
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted person object into the model's {@code Appointment} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted Appointment.
+     */
+    public Appointment toModelType(List<Doctor> allDoctors, List<Patient> allPatients) throws IllegalValueException {
+
+        if (id == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Integer.class.getSimpleName()));
+        }
+
+        Integer modelId = id;
+
+        if (doctorName == null) {
+            throw new IllegalValueException("Doctor name is missing");
+        }
+
+        Doctor modelDoctor = allDoctors.stream()
+                .filter(d -> d.getName().fullName.equals(doctorName))
+                .findFirst()
+                .orElseThrow(() -> new IllegalValueException("Doctor not found"));
+
+        if (patientName == null) {
+            throw new IllegalValueException("Patient name is missing");
+        }
+
+        Patient modelPatient = allPatients.stream()
+                .filter(p -> p.getName().fullName.equals(patientName))
+                .findFirst()
+                .orElseThrow(() -> new IllegalValueException("Patient not found"));
+
+        if (date == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Date.class.getSimpleName()));
+        }
+        if (!Date.isValidDate(date)) {
+            throw new IllegalValueException(Date.MESSAGE_CONSTRAINTS);
+        }
+        final Date modelDate = new Date(date);
+
+        if (time == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Time.class.getSimpleName()));
+        }
+        if (!Time.isValidTime(time)) {
+            throw new IllegalValueException(Time.MESSAGE_CONSTRAINTS);
+        }
+        final Time modelTime = new Time(time);
+
+        return new Appointment(modelId, modelDoctor, modelPatient, modelDate, modelTime);
```

Comment:
I think @sam-theman88 is removing the following two lines from the constructor of `Appointment` in his PR (see [this commit](https://github.com/AY2425S1-CS2103T-T09-3/tp/pull/89/commits/eb411b511f8cf0548a2cc2048aa740282e3d45f8)):

```java
        doctor.addAppointment(this);
        patient.addAppointment(this);
```

Do you need to call `addAppointment()` manually here if these two lines are removed?

Reply:
Discussed with @sam-theman88, we will add back the two lines and handle duplicates in the constructor of `Appointment` itself.